tests/test_load_fund_data.py

```diff
@@ -0,0 +1,238 @@
+    assert pandera_to_polars_dtype(pa.String()) == pl.String
+    assert pandera_to_polars_dtype(pa.Float()) == pl.Float64
+    assert pandera_to_polars_dtype(pa.Int()) == pl.Int64
+    assert pandera_to_polars_dtype(pa.Bool()) == pl.Boolean
+    assert pandera_to_polars_dtype(pa.DateTime()) == pl.Datetime
+    assert pandera_to_polars_dtype(pa.Object()) == pl.String
+
+
+def test_append_metadata_adds_fund_and_date():
+    """Test append_metadata function to ensure it adds fund name and date, as well as convert column name to screaming snake case."""
+    input_df = pl.DataFrame(
+        {
+            "FINANCIAL TYPE": ["Equity"],
+            "SYMBOL": ["AAPL"],
+            "SECURITY_NAME": ["Apple Inc."],
+            "SEDOL": [None],
+            "ISIN": ["US0378331005"],
+            "PRICE": [150.25],
+            "QUANTITY": [10.0],
+            "REALISED P/L": [50.0],
+            "MARKET VALUE": [1502.5],
+        }
+    )
+    with tempfile.NamedTemporaryFile(mode="w", suffix=".csv") as tmp_file:
+        input_df.write_csv(tmp_file.name)
+        fund_name = "FundA"
+        date = "2025-05-12"
+        result_df = append_metadata(tmp_file.name, fund_name, date)
+    assert "FUND" in result_df.columns
+    assert "DATE" in result_df.columns
+    assert "FINANCIAL_TYPE" in result_df.columns
+    assert "SYMBOL" in result_df.columns
+    assert "SECURITY_NAME" in result_df.columns
+    assert "SEDOL" in result_df.columns
+    assert "ISIN" in result_df.columns
+    assert "PRICE" in result_df.columns
+    assert "QUANTITY" in result_df.columns
+    assert "REALISED_P_L" in result_df.columns
+    assert "MARKET_VALUE" in result_df.columns
+    assert all(result_df["FUND"] == fund_name)
+    assert all(result_df["DATE"] == date)
+
+
+def test_load_data_to_db_successful():
+    """Test successful load_data_to_db function."""
+    mock_logger = MagicMock()
+    df = MagicMock(spec=pl.DataFrame)
+    database_path = "path/to/db.sqlite"
+    table_name = "fund_holdings"
+    with patch("logging.getLogger", return_value=mock_logger):
+        load_data_to_db(database_path, table_name, df)
+    expected_connection_string = f"sqlite:///{os.path.abspath(database_path)}"
+    df.write_database.assert_called_once_with(
+        connection=expected_connection_string,
+        table_name="fund_holdings",
+        if_table_exists="replace",
+    )
+
+
+def test_load_data_to_db_failure():
+    """Test failure scenario for load_data_to_db function."""
+    mock_logger = MagicMock()
+    df = MagicMock(spec=pl.DataFrame)
+    df.write_database.side_effect = Exception("Database write failed")
+    database_path = "path/to/db.sqlite"
+    table_name = "fund_holdings"
+    with (
+        pytest.raises(Exception, match="Database write failed"),
+        patch("logging.getLogger", return_value=mock_logger),
+    ):
+        load_data_to_db(database_path, table_name, df)
+    mock_logger.error.assert_called_once_with(
```

Comment:
This shows that it would be cleaner to allow the exception to escape, and test with `with pytest.raises`